client/state/sites/selectors.js

```diff
@@ -10,3 +10,23 @@
 export function getSite( state, siteId ) {
 	return state.sites.items[ siteId ] || null;
 }
+
+/**
+ * Returns the slug for a site, or null if the site is unknown.
+ *
+ * @param  {Object} state  Global state tree
+ * @param  {Number} siteId Site ID
+ * @return {String}        Site slug
+ */
+export function getSiteSlug( state, siteId ) {
+	const site = getSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+
+	if ( site.options && site.options.is_redirect ) {
+		return site.options.unmapped_url.replace( /^https?:\/\//, '' );
+	}
+
+	return site.URL.replace( /^https?:\/\//, '' ).replace( /\//g, '::' );
+}
```

Comment:
This differs from the current implementation in that `hasConflict` is not included:  https://github.com/Automattic/wp-calypso/search?q=hasConflict

More background at 5836-gh-calypso-pre-oss - looks like this is intended to handle fairly extreme edge cases around Jetpack and WP.com sites ending up with the same URL, and the current implementation would need some changes to fit with a reduxified sites list.  I personally think it would be ok to punt on it but wanted to mention.